### PR TITLE
Array_swaps_min_and_max.

### PR DESCRIPTION
**Pull request by ArteomKhromov**
Finding the maximum and minimum number in the entered array using methods.

### CODE REVIEW

File: task_DEV_2/task_DEV_2/Program.cs [hunk 1]
```diff
@@ -4,46 +4,57 @@ namespace task_DEV_2
...
-
-            if (arrayLength <= 20)
+            for (int i = 0; i < arrayLength; i++)
             {
-                int[] inputNumbers = new int[arrayLength];
+                Console.WriteLine($"Enter an array element № {i}");
+                inputNumbers[i] = Convert.ToInt32(Console.ReadLine());
+            }
+        }
+        static void ChangeMinMaxElements(int[] inputNumbers, int arrayLength)
```
Comment (MariaEroshevich) [inline task_DEV_2/task_DEV_2/Program.cs]: Please add empty line between the methods.

File: task_DEV_2/task_DEV_2/Program.cs [hunk 2]
```diff
@@ -4,46 +4,57 @@ namespace task_DEV_2
 {
     class Program
     {
-        static void Main(string[] args)
+        static void InputArrayElements(int[] inputNumbers, int arrayLength)
```
Comment (MariaEroshevich) [inline task_DEV_2/task_DEV_2/Program.cs]: Please change the name of the method and add return value.
static int[] CreateNumbersArray(int arrayLength)

File: task_DEV_2/task_DEV_2/Program.cs [hunk 3]
```diff
@@ -4,46 +4,57 @@ namespace task_DEV_2
...
+                    minIndex = i;
                 }
-                inputNumbers[maxIndex] = min;
-                inputNumbers[minIndex] = max;
+                Console.WriteLine(inputNumbers[i]);
+            }
+            inputNumbers[maxIndex] = min;
+            inputNumbers[minIndex] = max;
+        }
+        static void OutputArray(int[] inputNumbers, int arrayLength)
```
Comment (MariaEroshevich) [inline task_DEV_2/task_DEV_2/Program.cs]: Please add empty line between the methods.

File: task_DEV_2/task_DEV_2/Program.cs [hunk 4]
```diff
@@ -4,46 +4,57 @@ namespace task_DEV_2
...
+        {
+            for (int i = 0; i < arrayLength; i++)
+            {
+                Console.WriteLine(inputNumbers[i]);
+            }
+        }
+        static void Main(string[] args)
```
Comment (MariaEroshevich) [inline task_DEV_2/task_DEV_2/Program.cs]: Please add empty line between the methods.

Comment (MariaEroshevich) [reply task_DEV_2/task_DEV_2/Program.cs]: I suggest moving Main to the first place (to the beginning of the document)

File: task_DEV_2/task_DEV_2/Program.cs [hunk 5]
```diff
@@ -4,46 +4,57 @@ namespace task_DEV_2
...
 
-                for (int i = 0; i < arrayLength; i++)
-                {
-                    Console.WriteLine(inputNumbers[i]);
-                }
+            if (arrayLength <= 20)
+            {
+                int[] inputNumbers = new int[arrayLength];
+
+                InputArrayElements(inputNumbers, arrayLength);
```
Comment (MariaEroshevich) [inline task_DEV_2/task_DEV_2/Program.cs]: don't forget to add variable as this method  CreateNumbersArray will return the array.

File: task_DEV_2/task_DEV_2/Program.cs (hunk 1 above)
Comment (MariaEroshevich) [reply task_DEV_2/task_DEV_2/Program.cs]: Please change the name of the method and add return value.
static int[] SwapMinMaxElements(int[] inputNumbers, int arrayLength)

File: task_DEV_2/task_DEV_2/Program.cs [hunk 6]
```diff
@@ -4,46 +4,57 @@ namespace task_DEV_2
...
+                ChangeMinMaxElements(inputNumbers, arrayLength);
```
Comment (MariaEroshevich) [inline task_DEV_2/task_DEV_2/Program.cs]: don't forget to add variable as this method SwapMinMaxElements will return the array.

File: task_DEV_2/task_DEV_2/Program.cs (hunk 3 above)
Comment (MariaEroshevich) [reply task_DEV_2/task_DEV_2/Program.cs]: I suggest to rename it to PrintNumberArraryToConsole.

**Review by MariaEroshevich — CHANGES_REQUESTED** [review]
Change the name of pull request

File: task_DEV_2/task_DEV_2/Program.cs (hunk 1 above)
Comment (ArteomKhromov) [reply task_DEV_2/task_DEV_2/Program.cs]: done

File: task_DEV_2/task_DEV_2/Program.cs (hunk 2 above)
Comment (ArteomKhromov) [reply task_DEV_2/task_DEV_2/Program.cs]: done

File: task_DEV_2/task_DEV_2/Program.cs (hunk 3 above)
Comment (ArteomKhromov) [reply task_DEV_2/task_DEV_2/Program.cs]: done

File: task_DEV_2/task_DEV_2/Program.cs (hunk 4 above)
Comment (ArteomKhromov) [reply task_DEV_2/task_DEV_2/Program.cs]: done

File: task_DEV_2/task_DEV_2/Program.cs (hunk 5 above)
Comment (ArteomKhromov) [reply task_DEV_2/task_DEV_2/Program.cs]: done

File: task_DEV_2/task_DEV_2/Program.cs (hunk 6 above)
Comment (ArteomKhromov) [reply task_DEV_2/task_DEV_2/Program.cs]: done

File: task_DEV_2/task_DEV_2/Program.cs [hunk 7]
```diff
@@ -6,48 +6,65 @@ class Program
...
+                    min = inputNumbers[i];
+                    minIndex = i;
                 }
+                Console.WriteLine(inputNumbers[i]);
             }
-            else
+            inputNumbers[maxIndex] = min;
+            inputNumbers[minIndex] = max;
+
+            return inputNumbers[minIndex];
```
Comment (VadimBelkin) [inline task_DEV_2/task_DEV_2/Program.cs]: Why does this method return min element? This method has to return inputNumbers.

File: task_DEV_2/task_DEV_2/Program.cs [hunk 8]
```diff
@@ -6,48 +6,65 @@ class Program
...
-            Console.WriteLine("Enter the size of the array :");
+            Console.WriteLine("Enter the length of the array :");
             int arrayLength = Convert.ToInt32(Console.ReadLine());
 
             if (arrayLength <= 20)
             {
-                int[] inputNumbers = new int[arrayLength];
+                int[] inputNumbers = CreateNumbersArray(arrayLength);
+                PrintNumberArraryToConsole(inputNumbers, arrayLength);
+                SwapMinMaxElements(inputNumbers, arrayLength);
```
Comment (VadimBelkin) [inline task_DEV_2/task_DEV_2/Program.cs]: After swapping would be more logical do smth like this
inputNumbers = SwapMinMaxElements(inputNumbers, arrayLength);

**Review by VadimBelkin — CHANGES_REQUESTED** [review]

File: task_DEV_2/task_DEV_2/Program.cs (hunk 7 above)
Comment (ArteomKhromov) [reply task_DEV_2/task_DEV_2/Program.cs]: done

File: task_DEV_2/task_DEV_2/Program.cs (hunk 8 above)
Comment (ArteomKhromov) [reply task_DEV_2/task_DEV_2/Program.cs]: done

File: task_DEV_2/task_DEV_2/Program.cs [hunk 9]
```diff
@@ -6,48 +6,64 @@ class Program
...
         static void Main(string[] args)
         {
-            Console.WriteLine("Enter the size of the array :");
+            Console.WriteLine("Enter the length of the array :");
             int arrayLength = Convert.ToInt32(Console.ReadLine());
 
             if (arrayLength <= 20)
             {
-                int[] inputNumbers = new int[arrayLength];
+                int[] inputNumbers = CreateNumbersArray(arrayLength);
```
Comment (VadimBelkin) [inline task_DEV_2/task_DEV_2/Program.cs:14]: Why don't you print created array?

**Review by VadimBelkin — CHANGES_REQUESTED** [review]

File: task_DEV_2/task_DEV_2/Program.cs (hunk 9 above)
Comment (ArteomKhromov) [reply task_DEV_2/task_DEV_2/Program.cs:14]: done

**Review by VadimBelkin — APPROVED** [review]
Looks good for me. Wait for approve from Maria and can merge.

**Review by MariaEroshevich — APPROVED** [review]

**Pull request merged**